pkg/controller/machine/controller_test.go

```diff
@@ -681,9 +715,16 @@ func TestUpdateStatus(t *testing.T) {
 		},
 	}
 
+	// We don't need to recreate the test environment for every case.
+	g := NewWithT(t)
+	_, testEnv, err := StartEnvTest()
+	g.Expect(err).ToNot(HaveOccurred())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			g := NewWithT(t)
```

Comment:
Removing this means you now fail the entire suite instead of just one test case, notice the scoping of the `t` variable here vs moving this to line 719. You can just have it on both lines, or, avoid shadowing by calling the subtest `g` something else, `gs` is commonly used

Reply:
Ah ok - will go the `gs` route!